evm/src/get_challenges.rs

```diff
@@ -52,10 +52,11 @@ where
     C: GenericConfig<D, F = F>,
 {
     /// Computes all Fiat-Shamir challenges used in the STARK proof.
-    pub(crate) fn get_challenges<S: Stark<F, D>>(
+    pub(crate) fn get_challenges(
         &self,
         challenger: &mut Challenger<F, C::Hasher>,
-        stark: &S,
+        // Should correspond to `(stark.use_permutation_args(), stark.permutation_batch_size())`.
+        stark_permutation_info: (bool, usize),
```

Comment:
nit: I think it'd be slightly easier to read with separate `uses_permutation_args: bool, permutation_batch_size: usize`, then it'd be self-commenting

Reply:
Good point, done